lib/azure/armrest/storage/managed_storage_helper.rb

```diff
@@ -58,28 +58,46 @@ def get_blob_raw(disk_name, resource_group = configuration.resource_group, optio
     }
 
     # This call will give us an operations URL in the headers.
-    initial_url = build_url(resource_group, disk_name, 'BeginGetAccess')
-    response    = rest_post(initial_url, post_options.to_json)
+    begin_get_access_url      = build_url(resource_group, disk_name, 'BeginGetAccess')
+    begin_get_access_response = rest_post(begin_get_access_url, post_options.to_json)
+
     begin
-      headers = Azure::Armrest::ResponseHeaders.new(response.headers)
+      headers = Azure::Armrest::ResponseHeaders.new(begin_get_access_response.headers)
 
       # Using the URL returned from the above call, make another call that
       # will return the URL + SAS token.
       op_url = headers.try(:azure_asyncoperation) || headers.location
 
       unless op_url
         msg = "Unable to find an operations URL for #{disk_name}/#{resource_group}"
-        raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+        log('debug', "#{msg}: #{begin_get_access_response.headers}")
+        raise Azure::Armrest::NotFoundException.new(begin_get_access_response.code, msg, begin_get_access_response.body)
       end
 
       # Dig the URL + SAS token URL out of the response
       response = rest_get(op_url)
       body     = Azure::Armrest::ResponseBody.new(response.body)
```

Comment:
Jerry, couldn't we check the `body.status` here for "InProgress", and just loop until it's "Succeeded"?

Reply:
@djberg96 perhaps but I'm not sure if there is a 1-1 correlation between that status and the appearance of the  SAS URL token - do you know that for sure?